libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/alert/RestrictedAreaAlert.kt

```diff
@@ -0,0 +1,113 @@
+package com.mapbox.navigation.base.trip.model.alert
+
+import com.mapbox.geojson.Point
+
+/**
+ * Route alert type that provides information about restricted areas on the route.
+ *
+ * @see RouteAlert
+ * @see RouteAlertType.RestrictedArea
+ */
+class RestrictedAreaAlert private constructor(
+    metadata: Metadata,
+    coordinate: Point,
+    distance: Double,
+    alertGeometry: RouteAlertGeometry?
+) : RouteAlert<RestrictedAreaAlert.Metadata>(
+    RouteAlertType.RestrictedArea,
+    metadata,
+    coordinate,
+    distance,
+    alertGeometry
+) {
+
+    /**
+     * Transform this object into a builder to mutate the values.
+     */
+    fun toBuilder(): Builder = Builder(metadata, coordinate, distance).alertGeometry(alertGeometry)
+
+    /**
+     * Returns a string representation of the object.
+     */
+    override fun toString(): String {
+        return "RestrictedAreaAlert() ${super.toString()}"
+    }
+
+    /**
+     * Use to create a new instance.
+     *
+     * @see RestrictedAreaAlert
+     */
+    class Builder(
+        private val metadata: Metadata,
```

Comment:
What's the purpose of the Metadata?

Reply:
Metadata provides a type-safe set of information that's relevant only for one of the events, check to the example to see how easy it is to access without having to do any additional if-check that would otherwise be required if we just reused the Directions API response structure.

Some of the events do not have any additional information yet, but they might receive some in the future.

Reply:
But I'm not seeing the reason to have an extra Metadata class. Because the RestrictedAreaAlert is a type-safe set of information that's relevant only for the events.

Why not add the metadata contents to the class above?

Reply:
Let me try to do this locally and see how it plays out.

Reply:
Addressed, metadata might've been overkill, thanks for the suggestion!

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/3582#issuecomment-698435326 this outdated now / not necessary 😅